src/lib.rs

```diff
@@ -107,11 +107,18 @@ impl<T> Sender<T> {
     /// Sends a message asynchronously, if necessary waiting until enough
     /// capacity becomes available.
     pub async fn send(&mut self, message: T) -> Result<(), SendError<T>> {
-        let mut message = Some(message);
+        // Unless the previous send future was not polled to completion, there
+        // should be a cached notifier to take.
+        let notifier = self
+            .notifier
+            .take()
+            .unwrap_or_else(|| Box::new(Notifier::new()));
```

Comment:
You're storing the notifier in the future here, so the issue I reported shouldn't apply.

Reply:
Thanks for the review!